plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java

```diff
@@ -99,7 +99,7 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
             }
 
             // todo privileges should not be replaced for an alter
-            PrincipalPrivileges privileges = owner.isEmpty() && table.getOwner().isPresent() ? NO_PRIVILEGES : buildInitialPrivilegeSet(table.getOwner().get());
+            PrincipalPrivileges privileges = table.getOwner().map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
```

Comment:
Should this scenario be tested?

Reply:
I think this could only occur if a table is created in the metastore by some external process without an owner. I ran into this in a different scenario, but only due to modifying the code. The change is trivial enough that I don't think it warrants the extensive test setup required to reproduce.